catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/mappings/CswRecordMapperFilterVisitor.java

```diff
@@ -295,16 +361,42 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
                 Literal literal = getFactory(extraData).literal(orig);
                 return getFactory(extraData).before(other, literal);
             }
+        } else {
+            AttributeType type =
```

Comment:
There are ways to refactor this many-times duplicated block, but they are a bit messy.